vllm/config.py

```diff
@@ -2228,7 +2251,9 @@ def compute_hash(self) -> str:
         return hash_str
 
     def __post_init__(self):
-        if self.device == "auto":
+        if IS_IN_DOC_BUILD:
+            self.device_type = "cpu"
```

Comment:
Can we patch the `current_platform.device_type` on docs build to cpu instead of including the docs check in the code?

Reply:
I don't think so, `sphinx-argparse` is executing the parsers with `--help` in order to create the CLI documentation